ext/datastore/bigquery/adapter.go

```diff
@@ -84,6 +84,73 @@ func bqFieldModeTo(field BQField) (fieldMode, error) {
 	}
 	return fm, nil
 }
+func bqGoogleSheetsOptionsTo(m map[string]interface{}) *bqapi.GoogleSheetsOptions {
+	var skipLeadingRows int64
+	var sheetRange string
+
+	if val, ok := m["skip_leading_rows"]; ok {
+		skipLeadingRows = int64(val.(float64))
+	}
+
+	if val, ok := m["range"]; ok {
+		sheetRange = val.(string)
+	}
+
+	return &bqapi.GoogleSheetsOptions{
+		SkipLeadingRows: skipLeadingRows,
+		Range:           sheetRange,
+	}
+}
+
+func bqGoogleSheetsOptionsFrom(opt *bqapi.GoogleSheetsOptions) map[string]interface{} {
+	resultMap := make(map[string]interface{})
+
+	if opt.SkipLeadingRows != 0 {
+		// Map value of int has to be converted to float because of using interface{}
+		resultMap["skip_leading_rows"] = float64(opt.SkipLeadingRows)
+	}
+	if opt.Range != "" {
+		resultMap["range"] = opt.Range
+	}
+	return resultMap
+}
+
+func bqExternalDataConfigTo(es BQExternalSource) (*bqapi.ExternalDataConfig, error) {
+	var option bqapi.ExternalDataConfigOptions
+	var sourceType bqapi.DataFormat
+	switch bqapi.DataFormat(strings.ToUpper(es.SourceType)) {
+	case bqapi.GoogleSheets:
+		option = bqGoogleSheetsOptionsTo(es.Config)
+		sourceType = bqapi.GoogleSheets
+	default:
+		return &bqapi.ExternalDataConfig{}, fmt.Errorf("Source format not yet implemented %s", es.SourceType)
+	}
+
+	externalConfig := &bqapi.ExternalDataConfig{
+		SourceFormat: sourceType,
+		SourceURIs:   es.SourceURIs,
+		Options:      option,
+	}
+	return externalConfig, nil
+}
+
+func bqExternalDataConfigFrom(c *bqapi.ExternalDataConfig) (*BQExternalSource, error) {
+	var option map[string]interface{}
+
+	switch ExternalTableType(c.SourceFormat) {
+	case ExternalTableTypeGoogleSheets:
```

Comment:
Can't we use `bqapi.GoogleSheets` here as well like line number 122

Reply:
Agree, will refactor this